src/gallery-custom.tsx

```diff
@@ -42,6 +42,7 @@ export const CustomGallery: FC<CustomGalleryProps> = ({
   onOpen,
 }) => {
   const items = useRef(new Map<ItemRef, InternalItem>())
+  const pidRef = useRef(null)
```

Comment:
need a more expressive name for this

maybe `enqueuedForOpeningPidRef`?
or `openWhenReadyPidRef`?

or something more compact and still meaningful enough, at your discretion